src/packages/block/block-list/property-editors/block-list-editor/property-editor-ui-block-list.element.ts

```diff
@@ -93,47 +106,82 @@ export class UmbPropertyEditorUIBlockListElement extends UmbLitElement implement
 			// Notify that the value has changed.
 			//console.log('settings changed', this._value);
 		});
+		this.observe(this.#context.blockTypes, (blockTypes) => {
+			this._blocks = blockTypes;
+		});
 	}
 
-	#openBlockCatalogue() {
-		// Open modal.
+	async #openBlockCatalogue(view?: UmbBlockCatalogueView) {
+		//Open modal
+		const modalContext = this.#modalContext?.open(UMB_BLOCK_CATALOGUE_MODAL, {
+			data: { blocks: this._blocks ?? [], view },
+		});
+
+		const data = await modalContext?.onSubmit();
 
-		// TEMP Hack:
+		/**TODO: Insert next modal for data */
+		console.log('submitted', data);
 
-		const contentElementTypeKey = this.#context.getBlockTypes()[0]!.contentElementTypeKey;
+		if (!data) return;
 
-		const contentUdi = buildUdi('element', UmbId.new());
-		const settingsUdi = buildUdi('element', UmbId.new());
+		const block = this._blocks?.find((x) => x.contentElementTypeKey === data.key);
+
+		if (!block?.contentElementTypeKey) return;
 
 		this.#context.createBlock(
 			{
-				contentUdi,
-				settingsUdi,
+				contentUdi: buildUdi('element', UmbId.new()),
+				settingsUdi: buildUdi('element', UmbId.new()),
 			},
-			contentElementTypeKey,
+			block.contentElementTypeKey,
 		);
 	}
 
 	render() {
 		return html` ${repeat(
 				this._layouts,
-				(x) => x.contentUdi,
-				(layoutEntry) =>
-					html` <uui-button-inline-create></uui-button-inline-create>
-						<umb-property-editor-ui-block-list-block .layout=${layoutEntry}>
-						</umb-property-editor-ui-block-list-block>`,
+				(x, index) => x.contentUdi + index,
+				(layoutEntry, index) => html`
+					<umb-property-editor-ui-block-list-block .layout=${layoutEntry}> </umb-property-editor-ui-block-list-block>
+					${index !== this._layouts.length - 1 ? html`<uui-button-inline-create></uui-button-inline-create>` : nothing}
+				`,
```

Comment:
I'm not sure I understand this change. As we would like to have a Inline-create button above all Blocks. So you can create something at the top.

Reply:
It was just to make it match the current backoffice - I can revert it 😊